test/test_nn.py

```diff
@@ -9330,7 +9330,8 @@ def test_conv_noncontig_weights_and_bias(self, device):
             self.assertEqual(out1, out2)
 
     @unittest.skipIf(not TEST_LARGE_TENSOR, "not enough memory to run test")
-    @unittest.skipIf(sys.platform == "win32", "See https://github.com/pytorch/pytorch/issues/31650")
+    @allowCUDAOOM("This test fails with OOM on some systems, but succeeds on others with same memory."
```

Comment:
This approach seems dangerous. Let's say we increased our conv_transpose memory usage to, say, 128gb so that it always OOMs. Then the test would always get skipped and we'd never get signal that we did something wrong.

Reply:
@zou3519 I agree with your worry, but there is some decision that I have to make:

This is a test that requires a huge GPU memory, and it is hard to predict how much memory cudnn wants to use (for example it could be as large as 48GB). On my workstation, 16GB Volta + CUDA 10.2 there is no failure. But on Windows and on Power Linux, this test fails. So I have another worry that there will be some other combination of hardware + cudnn versions that also fails.

So I have to decide on how to write the policy for skipping this:
- If I skip as in this PR (allow OOM), then we would lose the ability to detect some kind of mistakes.
- If I skip by doing something like `skipIf(windows or power_linux or ...)`, it would be hard to write the condition, and this test could be very flaky and it is hard to know if there is some case that we are not aware of failing on some user's system

To be honest, I am now sure which sounds better.